pkg/apis/klum.cattle.io/v1alpha1/types.go

```diff
@@ -105,3 +108,37 @@ type NamedContext struct {
 	// Context holds the context information
 	Context Context `json:"context"`
 }
+
+// +genclient
+// +genclient:nonNamespaced
+// +k8s:deepcopy-gen:interfaces=k8s.io/apimachinery/pkg/runtime.Object
+
+type UserSyncGithub struct {
+	metav1.TypeMeta   `json:",inline"`
+	metav1.ObjectMeta `json:"metadata,omitempty"`
+	Spec              UserSyncGithubSpec `json:"spec"`
+	Status            UserSyncStatus     `json:"status,omitempty"`
+}
+
+type UserSyncGithubSpec struct {
+	User   string          `json:"user"`
+	Github *GithubSyncSpec `json:"github"`
```

Comment:
Now that this got its own type, you don't need to make this nullable. We could just assume/require that these fields are set.

You could then maybe simplify some of the validation logic by not checking for nil and such.

Reply:
Yeah, ideally I'd just delegate this to the validation webhook but since we don't have that yet I rather check everything for now.
If we change it I still need to check for empty value there.

Reply:
Not really. I'd say you could assume this field is set. If you remove the `*` and then only validate the fields inside `Github` then that's sufficient.

Maybe a different question to split this one up to other fields instead, but having them under `spec.github` does nicely separate it from the klum user reference

Reply:
Nevermind :)
simple change so here it is: [e6f6cad](https://github.com/jadolg/klum/pull/19/commits/e6f6cad0588a80d7277c99d3a3b7dd85d2948517)